src/main/java/com/ericsson/ei/notifications/EmailSender.java

```diff
@@ -48,17 +48,31 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(EmailSender.class);
 
     @Getter
-    @Value("${email.sender}")
-    private String sender;
+    @Value("${email.sender:noreply@ericsson.com}")
+    private static final String SENDER = "noreply@ericsson.com";
```

Comment:
The default value is now given 2 times, both as @Value and decleration. If Eclipse ot the IDE you are using is complaining about not declared variables then the @Value maybe will not work.

Reply:
A test must also ensure that a given value will not be set to default.

Reply:
After some googling I found that this will not work, a @Value from spring cannot change a final static variable, thus final static must be removed. And variable name changed to lower case :)

Reply:
yes If final static is used default values cant be changed. If that is the case required  ( where irrespective of user inputs, you want sender and subject as noreply.ericsson.com and Email Subscription Notification respectively.) ...these changes are suitable.

But what I have observed is, if user gives email and subject in application.proporties . Those will not override the sender and subject values in EmailSender class as they are set to final static. So we have to check it putting ifconditions ( if email.sender is empty then assign the value ) as I have done previously of my First push.

so please let me know for which case the changes are required so that accordingly I can proceed.

Reply:
The variable can't be static final then. Just type them out like this:
@Value("${email.sender:noreply@ericsson.com}")
private String sender;
That should make it possible for them to fetch the values from properties.
Also note that the sender variable does not need to be explicitly set, it is already set through the @Value annotation.

Reply:
Remove "ericsson" from the source code. change to some thing else or skip defaulting a value.
Let the Validation Properties discussion/issue solve it later.